src/magicproxy/async_proxy.py

```diff
@@ -110,12 +96,21 @@ def _clean_response_headers(headers):
             f"Disallowed by GitHub proxy. Allowed scopes: {', '.join(token_info.scopes)}"
         )
 
+    path = queries.clean_path_queries(_query_params_to_clean, path)
+
     return await _proxy_request(
         request=request,
         url=f"{GITHUB_API_ROOT}/{path}",
         headers={"Authorization": f"Bearer {token_info.github_token}"},
     )
 
+def queries_to_clean(querystrings: List[str]):
```

Comment:
I'm not sure I understand this redirection. Why's this needed? What's the expected use here?

Reply:
In a situation where someone puts the proxy behind an authentication layer (endpoints etc.), and the query string contains the API key, stripping it out before forwarding on to GitHub would be preferable.

I'm checking Cloud Endpoints now to see if they redact the key query parameter before forwarding to the backend. If so this might be unnecessary.

Reply:
What I mean is *how* would you call this method as part of your setup?

Reply:
Ah. As a part of the setup I could envision something like so


main.py
``` python
import argparse
from magicproxy import proxy
from magicproxy import magictoken

if __name__ == "__main__":
    parser = argparse.ArgumentParser("Magic GitHub Proxy")
    parser.add_argument('--port', metavar='P', type=int, nargs='?', default=5000,
                   help='the port to listen on')
    parser.add_argument('--debug',metavar='D', type=bool, nargs='?', default=False,
                        help='run in debug mode or not')

    args = parser.parse_args()
    
    proxy.keys = magictoken.Keys.from_env()
    proxy.queries_to_clean(['key'])
    proxy.app.run(host='0.0.0.0',port=args.port,debug=args.debug)
```

Reply:
I see. The sort of getter/setter thing is a bit weird in Python. I'd prefer you just made the module-level stuff public and modified it from main.py.

Reply:
Understood. Will do!